qml/components/TimeProgressBar.qml

```diff
@@ -86,23 +122,122 @@ Item {
         gradient: BarGradient { color: Style.timeBar.secondaryColor }
     }
 
-    Label {
-        id: text
+    RowLayout {
         anchors.fill: parent
+        spacing: 1
+
+        Item {
+            Layout.fillHeight: true
+            Layout.fillWidth: true
+            Label {
+                id: text
+                anchors.fill: parent
+                visible: !root.timeEditionInProgress
+                fontStyle: Style.timeBar.font
+
+                verticalAlignment: Text.AlignVCenter
+                horizontalAlignment: Text.AlignRight
+
+                elide: Text.ElideRight
+
+                text: d.formatTime(value)
+
+                BarTextGradient {
+                    source: parent
+                    value: Math.min(Math.max((progress.width - parent.x), 0.0) / width, 1.0)
+                }
+            }
+
+            LabelInput {
+                id: textEditableInput
+                anchors.fill: parent
+                visible: false
+
+                cursorVisible: true
+
+                verticalAlignment: Text.AlignVCenter
+                horizontalAlignment: Text.AlignRight
+
+                fontStyle: Style.spinBox.font
+                color: Style.highlightedApplicationColors[Style.mainColorIndex]
+
+                inputMask: "99:99:99"
+
+                onAccepted:
+                {
+                    timeValueChanged(d.deFormatTime(textEditableInput.text))
+                    visible = false
```

Comment:
Is `endTimeEdition()` anyhow called for this case? If not I think it should be called here to be consistent instead of manually setting visible to false.
Am I missing something?

Reply:
Done